server/repository/src/db_diesel/storage_connection.rs

```diff
@@ -218,6 +218,30 @@ impl StorageConnectionManager {
         con.batch_execute(sql)?;
         Ok(())
     }
+
+    pub fn sqlite_vacuum(&self, action: SqliteVacuumAction, setting: &SqliteVacuum) {
+        if cfg!(feature = "postgres") {
+            return;
+        }
+
+        let (should_vacuum, message) = match action {
+            SqliteVacuumAction::OnStartup => (setting.on_startup, "On Startup"),
+            SqliteVacuumAction::OnShutdown => (setting.on_shutdown, "On Shutdown"),
+            SqliteVacuumAction::AfterMigration => (setting.after_migration, "After Migration"),
+        };
+        if !should_vacuum {
+            return;
+        }
+
+        info!("{message} vacuum...");
+        self.execute("VACUUM").expect("Failed to vacuum");
```

Comment:
```suggestion
                match self.execute("VACUUM") {
            Ok(_) => {
                info!("VACUUM completed");
            }
            Err(e) => {
                error!("VACUUM failed: {e}");
            }
        }
```

I'm not sure if there's a reason it would fail other than out of disk space, but feels like we should probably try to keep going if it's just the vaccum that's failing?

Reply:
Thanks, updated